diagram-viewer/src/main/java/com/powsybl/diagram/viewer/MainViewController.java

```diff
@@ -302,6 +309,20 @@ private void initSubstationsTree(Network network) {
         vlTree.setShowRoot(true);
     }
 
+    private boolean testComponent(IdentifiableType type, Identifiable<?> identifiable) {
+        boolean result = false;
+        if (type == IdentifiableType.NETWORK || type == IdentifiableType.SUBSTATION || type == IdentifiableType.VOLTAGE_LEVEL) {
+            result = true;
+        } else {
+            if (identifiable instanceof Substation s) {
+                result = s.getVoltageLevelStream().anyMatch(v -> testComponent(type, v));
+            } else if (identifiable instanceof VoltageLevel v) {
+                result = v.getConnectableStream().anyMatch(c -> c.getType() == type);
```

Comment:
Does not work for switches nor buses. One would expect that selecting `BUS` item would remove the voltage levels with no bus in the bus view?